app/src/main/java/io/novafoundation/nova/app/root/navigation/governance/GovernanceNavigator.kt

```diff
@@ -45,14 +45,18 @@ class GovernanceNavigator(
 ) : BaseNavigator(navigationHolder), GovernanceRouter {
 
     override fun openReferendum(payload: ReferendumDetailsPayload) {
-        performNavigation(
-            cases = arrayOf(
-                R.id.mainFragment to R.id.action_open_referendum_details,
-                R.id.votedReferendaFragment to R.id.action_open_referendum_details,
-                R.id.referendaSearchFragment to R.id.action_open_referendum_details_from_referenda_search,
-            ),
-            args = ReferendumDetailsFragment.getBundle(payload)
-        )
+        // Close referendum details screen if it is already opened to open new one
+        // TODO: It's better to open new screen on top of the old one, but we need to consider that single shared state will be used for all screens
+        val currentDestination = navigationHolder.navController?.currentDestination
+        if (currentDestination?.id == R.id.referendumDetailsFragment) {
+            back()
+        }
```

Comment:
What about opening with replacement? We can create a separate action FROM referendumDetails TO referendumDetails. It will have the same effect but looks more clean to me